src/lib/pages/frontsite/Documents/DocumentDetails/DocumentCirculation/DocumentCirculation.js

```diff
@@ -21,13 +33,79 @@ class DocumentCirculation extends Component {
     return <LoanRequestForm document={documentDetails} />;
   };
 
+  myLoansButton = () => {
+    return (
+      <Link to={FrontSiteRoutes.patronProfile}>
+        <Button positive fluid>
+          My loans
+        </Button>
+      </Link>
+    );
+  };
+
+  renderRequestable = () => {
+    const {
+      documentDetails,
+      loans: { user_last_loan: lastLoan },
+    } = this.props;
+    return (
+      <>
+        <List>
+          {lastLoan && (
+            <List.Item>
+              <List.Icon name="info" />
+              <List.Content>
+                You loaned this document on <b>{toShortDate(lastLoan)}</b>
+              </List.Content>
+            </List.Item>
+          )}
+          {!_isEmpty(documentDetails.metadata.circulation) && (
+            <LoanAvailability
+              circulation={documentDetails.metadata.circulation}
+            />
+          )}
+        </List>
+        <AuthenticationGuard
+          authorizedComponent={this.renderLoanRequestForm}
+          loginComponent={this.loginToLoan}
+        />
+      </>
+    );
+  };
+
+  renderPendingRequest = () => {
+    return (
+      <>
+        <InfoMessage message="You have already requested a loan for this document." />
+        {this.myLoansButton()}
+      </>
+    );
+  };
+
+  renderOnLoan = () => {
+    return (
+      <>
+        <InfoMessage message="You have an ongoing loan for this document." />
+        {this.myLoansButton()}
+      </>
+    );
+  };
+
   render() {
     const {
       documentDetails,
       isLoading,
+      loans,
       loanRequestIsLoading,
       showTab,
+      hasRequestedLoan,
     } = this.props;
+    const {
+      user_has_loan: userHasLoan,
+      user_has_request: userHasRequest,
+    } = loans;
+    const isRequest = userHasRequest || hasRequestedLoan;
```

Comment:
sorry if I stress about naming, but I am not sure it is very clear the difference between `userHasRequest` and `hasRequestedLoan`: I guess it is previously requested or just requested?

I think @jrcastro2 is changing the field names, be aware.

What about:
```
const {
    has_active_loan: userHasActiveLoan,
    has_request: userHasPendingRequest
}
```
but I still not sure I understand why:
`const isRequest = userHasRequest || hasRequestedLoan;`
is needed.

What happens after putting a request on a literature? Let's see with the demo! :)

Reply:
`userHasRequest` comes from the `/me/loans` endpoint, while `hasRequestedLoan` is the result of clicking on the "Request" button. To find out in what UI state we are in we check if the user performed a request, otherwise fallback on the loans data. Perhaps there is a better way to do it.

I am aware about the potential name changes, it'll a quick change so don't worry about it.